uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/apis/leave-request-api.js

```diff
@@ -42,10 +42,12 @@ define([
         var deferred = $q.defer();
 
         if (!contactId || !periodId) {
-          deferred.resolve({
+          deferred.reject({
             is_error: 1,
             error_message: 'contact_id and period_id are mandatory'
           });
+
+          return deferred.promise;
```

Comment:
for now will keep it as it is, theres no harm in having 2 return statements as long as it makes code clean to understand